api/client/build.go

```diff
@@ -84,8 +86,37 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 		context  io.ReadCloser
 		isRemote bool
 		err      error
+		outBuf   io.Writer
 	)
 
+	// outBuf is for the Client's verbose output
+	// If script mode is set, then this output should be ignored
+	if !*scriptMode {
+		outBuf = cli.out
+	} else {
+		// First check whether the Docker daemon supports this feature.
+		serverResp, err := cli.call("GET", "/version", nil, nil)
```

Comment:
While I understand why you're doing this, I'm not sure its worth it or consistent with the rest of Docker. I can't think of any other check like this for other new flags. In general, there's no guarantee that newer CLIs will ever work with old servers w.r.t. new features, so its not worth adding this check.  Unless other @docker/maintainers want to start to add this check for other flags we may have added recently (and going forward).